libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -1444,7 +1444,7 @@ void PbftManager::pushSyncedPbftBlocksIntoChain() {
 
     if (pushPbftBlock_(std::move(period_data.first), std::move(period_data.second))) {
       LOG(log_dg_) << "Pushed synced PBFT block " << pbft_block_hash << " with period " << pbft_block_period;
-      net->setSyncStatePeriod(pbft_block_period);
+      net->setSyncStatePeriod(pbftSyncingPeriod());
```

Comment:
So the only thing this is used for is for setting deep_pbft_syncing flag and this is used for ignoring incoming packets when we are syncing. Packets are not even being sent to us while we are syncing. Packets are sent at the moment that the peer we are syncing from has send the last pbft packet. Because of the delay of this packet arriving to our node and being processed I think it is better to set this as soon as we received the packet and not only once we process it.

Even if we clear the queue on some error temporarily receiving future packets should not hurt us. This is something that is happening on startup anyhow when we are not sure if we are synced or not. I don't believe this can mess anything up.